### ISSUE TITLE
Shortcuts are not created automatically on some machines

### ISSUE THREAD

**Issue by Chuprinenko** [velopack/velopack#253]
### OS

Windows 10 Enterprise, os build 19045.4894

### Programming Language

net48

### VPK Version

vpk 0.0.626

### Library Version

Nuget 0.0.626

### What happened?

I'm packaging the version using --shortcuts "Desktop,StartMenu" param. But for some machines (seems virtual machines is the problem) this doesn't automatically create shortcuts when installing. The same story without specifying shortcuts param.

Nevertheless, it works ok when I add them in code:
 _new Velopack.Windows.Shortcuts().CreateShortcutForThisExe(ShortcutLocation.StartMenu | ShortcutLocation.Desktop);_

Seems nothing specific in logs.

### Relevant log output

```shell
2024-10-03 18:33:49> Information - Initializing WindowsVelopackLocator
2024-10-03 18:33:50> Information - Update.exe found in parent directory
2024-10-03 18:33:50> Information - Located valid manifest file at: C:\Users\user\AppData\Local\TEST_VELOPACK\current\sq.version
2024-10-03 18:33:50> Information - Starting Velopack App (Run).
2024-10-03 18:33:50> Information - Setting current process explicit AppUserModelID to 'velopack.TEST_VELOPACK'
2024-10-03 18:33:50> Information - Found fast exit hook: --veloapp-install
2024-10-03 18:33:50> Information - Completed hook, exiting...
2024-10-03 18:33:51> Information - Initializing WindowsVelopackLocator
2024-10-03 18:33:52> Information - Update.exe found in parent directory
2024-10-03 18:33:52> Information - Located valid manifest file at: C:\Users\user\AppData\Local\TEST_VELOPACK\current\sq.version
2024-10-03 18:33:52> Information - Starting Velopack App (Run).
2024-10-03 18:33:52> Information - Setting current process explicit AppUserModelID to 'velopack.TEST_VELOPACK'
```

**Comment by caesay** [comment]
Can you please provide the Velopack.log from setup/update? For setup, you can provide this by calling `setup.exe --log {fileLocation}`. For updates, Velopack.log is automatically created in your app dir.

Additionally, shortcuts are only created initially on setup.exe, or when you've added a new shortcut.

For example, if you pack `pack --shortcut StartMenuRoot` and then in a later update `pack --shortcut Desktop,StartMenuRoot`, the Desktop shortcut will be created. But, if the user deletes Desktop and then you pack a third update `pack --shortcut Desktop,StartMenuRoot`, then the Desktop shortcut will not be created again.

**Comment by Chuprinenko** [comment]
Sure.

[setup:19108] [19:04:33] [INFO] Starting Velopack Setup (0.0.626)
[setup:19108] [19:04:33] [INFO]     Location: "C:\\Projects\\Setup\\Client\\TEST_VELOPACK-win-Setup.exe"
[setup:19108] [19:04:33] [INFO]     Silent: false
[setup:19108] [19:04:33] [INFO]     Verbose: false
[setup:19108] [19:04:33] [INFO]     Log: Some("C:\\Projects\\logs.txt")
[setup:19108] [19:04:33] [INFO]     Install To: None
[setup:19108] [19:04:33] [INFO] OS: Windows 10.0.19045 (Windows 10 Enterprise) [64-bit], Arch=Some(
    X64,
)
[setup:19108] [19:04:33] [INFO] Reading bundle header...
[setup:19108] [19:04:33] [INFO] Bundle offset = 2508800, length = 351608845
[setup:19108] [19:04:33] [INFO] Loading bundle from embedded zip...
[setup:19108] [19:04:33] [INFO] Bundle loaded successfully.
[setup:19108] [19:04:33] [INFO] Reading package manifest...
[setup:19108] [19:04:33] [INFO] Package manifest loaded successfully.
[setup:19108] [19:04:33] [INFO]     Package ID: TEST_VELOPACK
[setup:19108] [19:04:33] [INFO]     Package Version: 9.4.1
[setup:19108] [19:04:33] [INFO]     Package Title: TEST_VELOPACK
[setup:19108] [19:04:33] [INFO]     Package Authors: Author
[setup:19108] [19:04:33] [INFO]     Package Description: TEST_VELOPACK
[setup:19108] [19:04:33] [INFO]     Package Machine Architecture: 
[setup:19108] [19:04:33] [INFO]     Package Runtime Dependencies: 
[setup:19108] [19:04:33] [INFO] Attempting to open global system mutex: 'velopack-TEST_VELOPACK'
[setup:19108] [19:04:33] [INFO] Checking application pre-requisites...
[setup:19108] [19:04:33] [INFO] Determining install directory...
[setup:19108] [19:04:33] [INFO] Installation Directory: "C:\\Users\\user\\AppData\\Local\\TEST_VELOPACK"
[setup:19108] [19:04:33] [INFO] There is 100.08 Gbit free space available at destination, this package requires 1.04 Gbit.
[setup:19108] [19:04:33] [INFO] Preparing and cleaning installation directory...
[setup:19108] [19:04:33] [INFO] Reading splash image...
[setup:19108] [19:04:33] [INFO] Starting installation!
[setup:19108] [19:04:33] [INFO] Showing splash screen immediately...
[setup:19108] [19:04:33] [INFO] Extracting Update.exe...
[setup:19108] [19:04:33] [INFO] Image is animated GIF (550x460), loading frames...
[setup:19108] [19:04:33] [INFO] Copying nupkg to packages directory...
[setup:19108] [19:04:34] [INFO] Successfully loaded 51 frames.
[setup:19108] [19:04:35] [INFO] Extracting 889 app files to 'C:\Users\user\AppData\Local\TEST_VELOPACK\current'...
[setup:19108] [19:04:45] [INFO] Creating shortcuts...
[setup:19108] [19:04:45] [INFO] Shortcut Previous Locations: ShortcutLocationFlags(0x0) (None)
[setup:19108] [19:04:45] [INFO] Shortcut Next Locations: ShortcutLocationFlags(START_MENU | DESKTOP) (Version { major: 9, minor: 4, patch: 1 })
[setup:19108] [19:04:45] [INFO] App Model ID: Some("velopack.TEST_VELOPACK")
[setup:19108] [19:04:45] [INFO] Searching for shortcuts containing root: 'C:\Users\user\AppData\Local\TEST_VELOPACK'
[setup:19108] [19:04:45] [WARN] Failed to update shortcuts: The system cannot find the file specified. (0x80070002)
[setup:19108] [19:04:45] [INFO] Starting process install hook
[setup:19108] [19:04:45] [INFO] Running --veloapp-install hook...
[setup:19108] [19:04:47] [INFO] Hook executed successfully (took 2749.6975ms)
[setup:19108] [19:04:47] [INFO] Checking for running processes in: C:\Users\user\AppData\Local\TEST_VELOPACK
[setup:19108] [19:04:47] [INFO] Writing uninstall registry key...
[setup:19108] [19:04:47] [INFO] Starting app...
[setup:19108] [19:04:47] [INFO] About to launch: 'C:\Users\user\AppData\Local\TEST_VELOPACK\current\test.exe' in dir 'C:\Users\user\AppData\Local\TEST_VELOPACK\current'
[setup:19108] [19:04:47] [INFO] Args: CommandArgs { inner: [] }
[setup:19108] [19:04:47] [INFO] Installation completed successfully!
